airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/state/CheckpointManager.kt

```diff
@@ -173,14 +189,48 @@ abstract class StreamsCheckpointManager<T, U>() :
                         streamCheckpoints.remove(index)
                             ?: throw IllegalStateException("Checkpoint not found for index: $index")
                     log.info { "Flushing checkpoint for stream: $stream at index: $index" }
-                    val outMessage = outputFactory.from(checkpointMessage)
-                    outputConsumer.accept(outMessage)
+                    sendMessage(checkpointMessage)
                 } else {
                     break
                 }
             }
         }
     }
+
+    private suspend fun sendMessage(checkpointMessage: T) =
+        withContext(Dispatchers.IO) {
+            lastFlushTimeMs.set(timeProvider.currentTimeMillis())
+            val outMessage = outputFactory.from(checkpointMessage)
+            outputConsumer.accept(outMessage)
+        }
+
+    override fun getLastSuccessfulFlushTimeMs(): Long {
+        if (flushLock.isLocked) {
```

Comment:
Yes. That's technically not correct though, because the flusher might have nothing to flush. It's probably better just to take the lock to return the time.

There's some tricky things like that: what if I publish a flush update event with no upcoming checkpoints? or checkpoints for just one stream? Maybe if I waited a minute I'd have more to force? Etc. This feels like best-effort tho.